pkg/apiserver/user/sso/service.go

```diff
@@ -273,23 +288,35 @@ func (s *Service) discoverOIDC(issuer string) (*oidcWellKnownConfig, error) {
 	ctx, cancel := context.WithTimeout(s.lifecycleCtx, discoveryTimeout)
 	defer cancel()
 
-	wellKnownURL := strings.TrimSuffix(issuer, "/") + "/.well-known/openid-configuration"
+	wellKnownURL := issuer + "/.well-known/openid-configuration"
 	resp, err := resty.New().R().SetContext(ctx).SetResult(&oidcWellKnownConfig{}).Get(wellKnownURL)
 	if err != nil {
 		return nil, ErrDiscoverFailed.Wrap(err, "Failed to discover OIDC endpoints")
 	}
 	wellKnownConfig := resp.Result().(*oidcWellKnownConfig)
-	if wellKnownConfig.Issuer != issuer {
+	if strings.TrimSuffix(wellKnownConfig.Issuer, "/") != issuer {
 		return nil, ErrDiscoverFailed.New("Issuer did not match in the OIDC provider, expect %s, got %s", issuer, wellKnownConfig.Issuer)
 	}
 	if len(wellKnownConfig.TokenURL) == 0 {
-		return nil, ErrDiscoverFailed.New("TokenURL is not provided in the OIDC provider")
+		return nil, ErrDiscoverFailed.New("token_endpoint is not provided in the OIDC provider")
```

Comment:
That's we validate many fields that must be implemented in the specification, `OpenID Provider Configuration Validation` only mentions `issuer url` and `iss Claim`?

And what is the `iss Claim value in ID Tokens issued`

Reply:
https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata mentions the required metadata.

Reply:
https://github.com/coreos/go-oidc
https://connect2id.com/blog/how-to-validate-an-openid-connect-id-token

It looks like we should verify the issuer and issuer claims from id tokens in the exchange phase, not in the discovery phase.

Reply:
Maybe we need to do the verification here:

https://github.com/pingcap/tidb-dashboard/blob/master/pkg/apiserver/user/sso/service.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L362